src/fal/planner/schedule.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Iterator
+
+import networkx as nx
+
+from fal.node_graph import DbtModelNode, NodeGraph, NodeKind
+from fal.planner.tasks import SUCCESS, DBTTask, FalHookTask, FalModelTask, Node
+
+
+def create_node(
+    node: str | nx.DiGraph, properties: dict, node_graph: NodeGraph
+) -> Node:
+    kind = properties["kind"]
+    if isinstance(node, nx.DiGraph):
+        model_ids = sorted(
+            list(node),
+            key=lambda node: node == properties["exit_node"],
+        )
+    else:
+        model_ids = [node]
+
+    if kind is NodeKind.DBT_MODEL:
+        task = DBTTask(model_ids=model_ids)
+    else:
+        assert kind is NodeKind.FAL_MODEL
+        task = FalModelTask(model_ids=model_ids)
+
+    model_node = node_graph.get_node(model_ids[-1])
+    assert model_node is not None
+
+    if isinstance(task, FalModelTask):
+        assert isinstance(model_node, DbtModelNode)
+        task.bound_model = model_node.model
```

Comment:
couldn't this happen in the constructor?

Reply:
Ah seems like I can simply lift the `model_node` retrieval to a bit above, seems to be solved now!